src/liballoc/rc.rs

```diff
@@ -1702,14 +1700,18 @@ impl<T> Weak<T> {
     /// This can be used to safely get a strong reference (by calling [`upgrade`]
     /// later) or to deallocate the weak count by dropping the `Weak<T>`.
     ///
-    /// It takes ownership of one weak count. In case a [`null`] is passed, a dangling [`Weak`] is
-    /// returned.
+    /// It takes ownership of one weak count (with the exception of pointers created from [`new`],
+    /// these don't have corresponding weak count).
```

Comment:
I'd either write "created from `new`; these" or "created from `new`, as"; the current wording is a comma splice.